utils/retrieval_evaluator.py

```diff
@@ -0,0 +1,559 @@
+            f.write("important_entries: The number of unique queries evaluated in this assessment.\n")
+
+            f.write("\nSummary:\n")
+            for key, value in summary.items():
+                f.write(f"{key}: {value:.2f}\n")
+
+            f.write("\nEnhanced Summary:\n")
+            f.write(f"Overall Recall Score: {enhanced_summary['overall_recall_score']:.2f}\n")
+            f.write(f"Overall Irrelevant %: {enhanced_summary['overall_irrelevant_percent']:.2f}%\n")
+            f.write(f"Overall Trivia %: {enhanced_summary['overall_trivia_percent']:.2f}%\n")
+            f.write(f"Scored Questions Recall: {enhanced_summary['scored_questions_recall']:.2f}\n")
+            f.write(f"Total Entries: {enhanced_summary['total_entries']}\n")
+            f.write(f"Total Scored Entries: {enhanced_summary['total_scored_entries']}\n")
+            f.write(f"Total Memories: {enhanced_summary['total_memories']}\n")
+            f.write(f"Average Memories per Entry: {enhanced_summary['average_memories_per_entry']:.2f}\n")
+
+            f.write("\nTest-specific Summaries:\n")
+            for test_summary in enhanced_summary['test_summaries']:
+                f.write(f"\n{test_summary['test']}:\n")
+                f.write(f"  Recall Score: {test_summary['recall_score']:.2f}\n")
+                f.write(f"  Irrelevant %: {test_summary['irrelevant_percent']:.2f}%\n")
+                f.write(f"  Trivia %: {test_summary['trivia_percent']:.2f}%\n")
+                f.write(f"  Scored Questions: {test_summary['is_scored_question']}\n")
+
+            f.write("\nRetrieved vs Filtered Memories Summary:\n")
+            rf_summary = enhanced_summary['retrieved_filtered_summary']
+            f.write(f"Mean Retrieved Memories: {rf_summary['mean_retrieved']:.2f} (SD: {rf_summary['sd_retrieved']:.2f})\n")
+            f.write(f"Mean Filtered Memories: {rf_summary['mean_filtered']:.2f} (SD: {rf_summary['sd_filtered']:.2f})\n")
+            f.write(f"Mean Retrieval Efficiency: {rf_summary['mean_efficiency']:.2f} (SD: {rf_summary['sd_efficiency']:.2f})\n")
+            f.write(f"Total Retrieved Memories: {rf_summary['total_retrieved']}\n")
+            f.write(f"Total Filtered Memories: {rf_summary['total_filtered']}\n")
+            f.write(f"Overall Retrieval Efficiency: {rf_summary['overall_efficiency']:.2f}\n")
+
+            retrieved = [r['retrieved_memories_count'] for r in results]
+            filtered = [r['filtered_memories_count'] for r in results]
+            t_stat, p_value = stats.ttest_rel(retrieved, filtered)
+            f.write(f"\nPaired t-test results (Retrieved vs Filtered):\n")
+            f.write(f"t-statistic: {t_stat:.4f}\n")
+            f.write(f"p-value: {p_value:.4f}\n")
+
+    def capture_comparison_data(self, query: str, retrieved_memories: List[Union[Dict[str, Any], Any]], filtered_memories: List[Union[Dict[str, Any], Any]]):
+        comparison_data = {
```

Comment:
Don't know if you can make it work, but the current benchmark uses ~3 queries to get the retrieved memories. Are you able to adapt the query string to a list?

Right now I am just `"\n".join()`-ing the queries.